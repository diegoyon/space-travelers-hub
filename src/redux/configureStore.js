import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missions';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
