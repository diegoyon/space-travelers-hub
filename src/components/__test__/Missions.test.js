import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import renderer from 'react-test-renderer';
import MissionLi from '../Missions';

describe(MissionLi, () => {
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MissionLi
          key={1}
          id="1"
          name="Mission impossible"
          description="An impossible mission"
          reserved
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('mission displays correct name information', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MissionLi
          key={1}
          id="1"
          name="Mission impossible"
          description="An impossible mission"
          reserved
        />
      </Provider>,
    );
    const missionName = getByTestId('name').textContent;
    expect(missionName).toBe('Mission impossible');
  });

  it('mission displays correct description information', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <MissionLi
          key={1}
          id="1"
          name="Mission impossible"
          description="An impossible mission"
          reserved
        />
      </Provider>,
    );
    const missionDescription = getByTestId('description').textContent;
    expect(missionDescription).toBe('An impossible mission');
  });
});
