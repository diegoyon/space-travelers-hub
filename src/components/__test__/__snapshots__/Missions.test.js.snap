// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionLi renders correctly 1`] = `
<li
  className="mission-li"
>
  <p
    className="mission-name"
    data-testid="name"
  >
    Mission impossible
  </p>
  <p
    className="mission-description"
    data-testid="description"
  >
    An impossible mission
  </p>
  <div
    className="mission-status"
  >
    <p
      className="joined"
    >
      member
    </p>
  </div>
  <div
    className="btn-container joined"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      leave
    </button>
  </div>
</li>
`;
